daemon/api.go

```diff
@@ -268,6 +268,11 @@ func loginUser(c *Command, r *http.Request, user *auth.UserState) Response {
 		loginData.Email = loginData.Username
 		loginData.Username = ""
 	}
+
+	if loginData.Email == "" && user.Email != "" {
```

Comment:
Right, I've got a working test now. Sadly there's quite some duplication in the api_test tests for this. It's probably worth revisiting them and doing the table thing when there's some time.